src/mlpack/bindings/markdown/replace_all_copy.hpp

```diff
@@ -0,0 +1,32 @@
+/**
+ * @file bindings/markdown/replace_all_copy.hpp
+ * @author Nippun Sharma
+ *
+ * Replace a substring in a string.
+ * 
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_BINDINGS_MARKDOWN_REPLACE_ALL_COPY_HPP
+#define MLPACK_BINDINGS_MARKDOWN_REPLACE_ALL_COPY_HPP
+
+#include <mlpack/prereqs.hpp>
+
+// Replaces all occurences of "from" in "str" to "to".
+inline std::string replace_all_copy(const std::string& str,
```

Comment:
Awesome, thanks for the custom implementation here.  I know it is nitpicky, but can you change the names to match the mlpack naming schemes?  `ReplaceAllCopy()`, `strCopy`, `startPos`, etc. :+1: